docs/programmable-control-flow.md

```diff
@@ -0,0 +1,145 @@
+# Programmable Control Flow
+
+New language features:
+
+* Lambda parameters can be values, which turns the lambda into a case.
```

Comment:
What's the rule for deciding whether a parameter is a value or a variable binder?  That is, when is `{ x => ... }` testing equality against `x` and when is it introducing `x`?

Reply:
I've added a note here. The rule is: If it can be a parameter, it is. Otherwise, it is interpreted as a pattern.